### PR TITLE
feat(batch-calls): add batch calls

### PR DESCRIPTION
**Pull request by kkotelczuk**

### CODE REVIEW

File: src/data.js [hunk 1]
```diff
@@ -20,6 +20,43 @@ class Data extends QueryBuilder {
     return query ? `${url}?${query}` : url
   }
 
+  _batchBodyBuilder(body) {
+    const {instanceName, className} = this.instance
+    const batchBody = {requests: []}
+
+    body.forEach(item => {
```
Comment (Idered) [inline src/data.js]: Try to use `reduce` instead of `forEach`.

File: src/data.js [hunk 2]
```diff
@@ -375,12 +434,19 @@ class Data extends QueryBuilder {
    * @returns {Promise}
    *
    * @example {@lang javascript}
-   * data.posts.delete(55)
+   * data.posts.delete(55) || data.posts.delete([55, 56, 57])
```
Comment (Idered) [inline src/data.js]: Put example in new line :P

File: tests/unit/data.js [hunk 3]
```diff
@@ -283,6 +283,22 @@ describe('Data', () => {
...
+        {name: 'John'},
+        {name: 'Joe'}
+      ]
+
+      api.post(`/v1/instances/${instanceName}/batch/`)
+        .query(users)
+        .reply(200, users)
+
+      data.users.create(users).then(object => {
+        should(object).be.Array()
```
Comment (Idered) [inline tests/unit/data.js]: change `object` var name to `items`.

File: tests/unit/data.js [hunk 4]
```diff
@@ -283,6 +283,22 @@ describe('Data', () => {
...
+        should(object).have.property('name').equal('John')
```
Comment (Idered) [inline tests/unit/data.js]: `object` is array of objects. Use https://shouldjs.github.io/#assertion-propertybypath to make comparison.

File: src/data.js [hunk 5]
```diff
@@ -376,11 +435,19 @@ class Data extends QueryBuilder {
    *
    * @example {@lang javascript}
    * data.posts.delete(55)
+   * data.posts.delete([55, 56, 57])
```
Comment (mkucharz) [inline src/data.js]: As far as I remember some of the things like this are possible to get also using one request and query e.g. (id _in [55, 56,57]). I'm not saying that batch here is wrong... @23doors am I right?

File: tests/unit/data.js [hunk 6]
```diff
@@ -283,6 +283,25 @@ describe('Data', () => {
...
+      ]
+
+      api.post(`/v1/instances/${instanceName}/batch/`)
+        .query(users)
+        .reply(200, [
+          {content: {name: 'John'}, code: 201},
+          {content: {name: 'Joe'}, code: 201}
+        ])
+
+      data.users.create(users).then(items => {
```
Comment (Idered) [inline tests/unit/data.js]: should be `users.create(....`. Won't work on v1

File: tests/e2e/data.js [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+
+  it('can create multiple objects', function (done) {
+    data.tests
+      .create([
+        {test: 'batch1', test2: 'secret'},
+        {test: 'batch2', test2: 'secret'},
+        {test: 'batch3', test2: 'secret'},
+        {test: 'batch4', test2: 'secret'}
+      ])
+      .then(() => {
```
Comment (Idered) [inline tests/e2e/data.js]: `.then(done)` - that should be enough :)

**Pull request merged**